sklearn/preprocessing/tests/test_imputation.py

```diff
@@ -357,3 +362,70 @@ def test_imputation_copy():
 
     # Note: If X is sparse and if missing_values=0, then a (dense) copy of X is
     # made, even if copy=False.
+
+
+def test_missing_indicator():
+    X1_orig = np.array([
+             [-1,  -1,   1,   3],
+             [4,  -1,   0,  -1],
+             [8,  -1,   1,  0],
+             [0,  -1,   0,  15],
+             [16,  -1,   1,  19]
+    ])
+    X2_orig = np.array([
+             [5,  1,   1,   -1],
+             [-1,  -1,   2,  3],
+             [2,  3,   4,  0],
+             [0,  -1,   5,  -1],
+             [11,  -1,   1,  1]
+    ])
+
+    for X1, X2, missing_values in [(X1_orig, X2_orig, -1),
+                                   (X1_orig + 1, X2_orig + 1, 0)]:
+        mask = X2 == missing_values
+        expect_feat_missing = np.where(np.any(X1 == missing_values, axis=0))[0]
+        for retype in [np.array, sparse.csr_matrix,
+                       sparse.csc_matrix, sparse.lil_matrix]:
+            # features = "train":
+            MI = MissingIndicator(missing_values=missing_values)
+            MI.fit(retype(X1))
+            X2_tr = MI.transform(X2)
+            features = MI.feat_with_missing_
+            assert_array_equal(expect_feat_missing, features)
+            assert_array_equal(np.asarray(X2_tr), mask[:, features])
```

Comment:
yeah yeah sorry it is array or sparse matrix. I was asking about the equality between the sparse matrices because X2_tr will be sparse while mask is array. Is it okay to densify and use ``assert_array_equals`` ?